storage/storage.go

```diff
@@ -36,40 +38,81 @@ type ETLSource struct {
 
 // Next reads the next test object from the tar file.
 // Returns io.EOF when there are no more tests.
-func (rr *ETLSource) NextTest() (string, []byte, error) {
+func (rr *ETLSource) NextTest() (string, []byte, int, error) {
```

Comment:
At some point the size of the tuple becomes large enough that it should be a struct. Depending on your sense of taste, that point is either 3, 4, or 5 elements. I'll leave the choice up to you, but returning a 4-element tuple does not feel good. I think my preference would be returning a `(Test, error)` tuple at this point.